pkg/ccl/backupccl/system_schema.go

```diff
@@ -103,6 +113,114 @@ func jobsRestoreFunc(
 ) error {
 	executor := execCfg.InternalExecutor
 
+	const (
+		statesToRevert = `('` + string(jobs.StatusRunning) + `', ` +
+			`'` + string(jobs.StatusPauseRequested) + `', ` +
+			`'` + string(jobs.StatusPaused) + `')`
+	)
+
+	err := func() (err error) {
+		// makeJobsList writes the given jobs as a SQL tuple set to the
+		// given queryBuilder.
+		addJobsListTuple := func(jobIDs []int64, queryBuilder *strings.Builder) {
+			fmt.Fprint(queryBuilder, "(")
+			for i, job := range jobIDs {
+				if i > 0 {
+					fmt.Fprint(queryBuilder, ", ")
+				}
+				fmt.Fprintf(queryBuilder, "'%d'", job)
+			}
+			fmt.Fprint(queryBuilder, ")")
+		}
+
+		jobsToRevert := make([]int64, 0)
+		query := `SELECT id, payload FROM ` + tempTableName + ` WHERE status IN ` + statesToRevert
+		it, err := executor.QueryIteratorEx(
+			ctx, "restore-fetching-job-payloads", txn,
+			sessiondata.InternalExecutorOverride{User: security.RootUserName()},
+			query)
+		if err != nil {
+			return errors.Wrap(err, "fetching job payloads")
+		}
+		defer func() {
+			closeErr := it.Close()
+			if err == nil {
+				err = closeErr
+			}
+		}()
+		for {
+			ok, err := it.Next(ctx)
+			if !ok {
+				if err != nil {
+					return err
+				}
+				break
+			}
+
+			r := it.Cur()
+			id, payloadBytes := r[0], r[1]
+			rawJobID, ok := id.(*tree.DInt)
+			if !ok {
+				return errors.Errorf(
+					"job: failed to read job id as DInt (was %T)", id)
+			}
+			jobID := int64(*rawJobID)
+
+			payload, err := jobs.UnmarshalPayload(payloadBytes)
+			if err != nil {
+				return errors.Wrap(err, "failed to unmarshal job to restore")
+			}
+			if payload.Type() == jobspb.TypeSchemaChange {
+				details, ok := payload.UnwrapDetails().(jobspb.SchemaChangeDetails)
+				if !ok {
+					return errors.Newf("expected schema change job to have schema change details, got %t",
+						payload.UnwrapDetails())
+				}
+				if len(payload.DescriptorIDs) != 1 {
+					continue
+				}
+				descriptorID := payload.DescriptorIDs[0]
+				desc, err := catalogkv.GetDescriptorByID(ctx, txn, execCfg.Codec, descriptorID)
+				if err != nil {
+					return err
+				}
+				table, ok := desc.(catalog.TableDescriptor)
+				if !ok {
+					continue
+				}
+				allSpans := table.PrimaryIndexSpan(execCfg.Codec)
+				for i := range details.ResumeSpanList {
+					details.ResumeSpanList[i] = jobspb.ResumeSpanList{ResumeSpans: []roachpb.Span{allSpans}}
+				}
+				payload.Details = jobspb.WrapPayloadDetails(details)
+
+				payloadBytes, err := protoutil.Marshal(payload)
+				if err != nil {
+					return err
+				}
+				q := fmt.Sprintf("UPDATE %s SET payload = $1 WHERE id = $2", tempTableName)
+				if _, err := executor.Exec(ctx, "updating-job-progress", txn, q, payloadBytes, jobID); err != nil {
```

Comment:
I'm a little nervous this txn may never complete? all the other nodes are running pollers on the jobs table so I worry if this will just get pushed every time when trying to update a giant jobs table. maybe if we add `for update` to the select above? I'm curious what @ajwerner thinks

Reply:
SFU seems like a good idea. Especially given we really don't want the registries trying to adopt jobs anyway.

Reply:
it occurs to me that we wrote the jobs in this txn too, so that is actually moot I guess?

Reply:
Turns out nobody is reading this

Reply:
I did want to check what the expected semantics are here wrt the txn use between the select above and this update given the new streaming internal executor. Is it safe to update this data while iterating over the streamed results?